content/articles/leandrocfe-navigating-filament-pages-with-previous-and-next-buttons.md

```diff
@@ -0,0 +1,191 @@
+            ->outlined()
+            ->tooltip("Next {$this->getRecordTitle()}");
+    }
+}
+```
+
+## Configuring Actions in Pages
+
+Next, we extend the capabilities of the ViewRecord and EditRecord classes by adding a `CanPaginateViewRecord` trait. This trait, residing in the `app/Filament/Resources/Pages/Concerns` folder, configures actions for pagination and provides methods to retrieve previous and next records based on the current record.
+
+```php
+namespace App\Filament\Resources\Pages\Concerns;
+
+use App\Filament\Resources\Actions\NextAction;
+use App\Filament\Resources\Actions\PreviousAction;
+use Filament\Actions\Action;
+use Filament\Actions\DeleteAction;
+use Filament\Actions\EditAction;
+use Filament\Actions\ForceDeleteAction;
+use Filament\Actions\ReplicateAction;
+use Filament\Actions\RestoreAction;
+use Illuminate\Database\Eloquent\Model;
+
+trait CanPaginateViewRecord
+{
+    protected function configureAction(Action $action): void
+    {
+        $this->configureActionRecord($action);
+
+        match (true) {
+            $action instanceof DeleteAction => $this->configureDeleteAction($action),
+            $action instanceof EditAction => $this->configureEditAction($action),
+            $action instanceof ForceDeleteAction => $this->configureForceDeleteAction($action),
+            $action instanceof ReplicateAction => $this->configureReplicateAction($action),
+            $action instanceof RestoreAction => $this->configureRestoreAction($action),
+
+            //new actions
+            $action instanceof PreviousAction => $this->configurePreviousAction($action),
+            $action instanceof NextAction => $this->configureNextAction($action),
+            
+            default => null,
+        };
+    }
+
+    protected function configurePreviousAction(Action $action): void
+    {
+        if ($this->getPreviousRecord()) {
+            $action->url(fn (): string => static::getResource()::getUrl(static::getResourcePageName(), ['record' => $this->getPreviousRecord()]));
+        } else {
+            $action
+                ->disabled()
+                ->color('gray');
+        }
+    }
+
+    protected function configureNextAction(Action $action): void
+    {
+        if ($this->getNextRecord()) {
+            $action->url(fn (): string => static::getResource()::getUrl(static::getResourcePageName(), ['record' => $this->getNextRecord()]));
+        } else {
+            $action
+                ->disabled()
+                ->color('gray');
+        }
+    }
+
+    protected function getPreviousRecord(): ?Model
+    {
+        return $this
+            ->getRecord()
+            ->where('id', '<', $this->getRecord()->id)
+            ->orderBy('id', 'desc')
+            ->first();
+    }
+
+    protected function getNextRecord(): ?Model
+    {
+        return $this
+            ->getRecord()
+            ->where('id', '>', $this->getRecord()->id)
+            ->orderBy('id', 'asc')
+            ->first();
+    }
+}
```

Comment:
May be worth mentioning that these, as written, only work when the expected sort is using orderable IDs (auto-incrementing IDs, ULIDs, etc.). I don't think you'd need to add examples of other ways to do this, but for the people who are likely to just copy-paste the code, it's probably a worthwhile callout.

Reply:
Hi Alex, sorry for the delay. I added this note.